DTAdmin/Result/ResultByGroupViewController.swift

```diff
@@ -0,0 +1,135 @@
+        resultsTableView.refreshControl = refreshControl
+        
+    }
+    
+    private func prepareView(group: GroupStructure) {
+        self.startActivity()
+        self.title = NSLocalizedString("Results by \(group.groupName)", comment: "Title for results table list view")
+        self.refreshControl.beginRefreshing()
+        self.loadDataFromAPI()
+    }
+    
+    @objc private func loadDataFromAPI() {
+        guard let groupId = self.group?.groupId else { return }
+        var testsIds = [String]()
+        var subjectsIds = [String]()
+        
+        DataManager.shared.getResultTestIds(byGroup: groupId) { (error, testIds) in
+            if let error = error {
+                self.stopActivity()
+                self.showWarningMsg(error)
+            } else {
+                guard let testIds = testIds else {
+                    self.stopActivity()
+                    self.showWarningMsg("Error occured during data handle")
+                    return
+                }
+                for i in testIds {
+                    testsIds.append(i["test_id"]!)
+                }
+                
+                DataManager.shared.getTestsBy(ids: testsIds, completionHandler: { (error, tests) in
+                    if let error = error {
+                        self.stopActivity()
+                        self.showWarningMsg(error)
+                    } else {
+                        guard var tests = tests else {
+                            self.showWarningMsg("Wrong API response")
+                            self.stopActivity()
+                            return
+                        }
+                        self.tests = tests
+                        for i in tests {
+                            subjectsIds.append(i.subjectId)
+                        }
+                        
+                        DataManager.shared.getSubjectsBy(ids: subjectsIds, completionHandler: { (error, subjects) in
+                            for i in 0..<tests.count {
+                                let testSubj = subjects?.filter({ $0.id == tests[i].subjectId })
+                                tests[i].subjectName = testSubj?.first?.name
+                            }
+                            DispatchQueue.main.async {
+                                self.tests = tests
+                                self.resultsTableView.reloadData()
+                                self.refreshControl.endRefreshing()
+                                self.stopActivity()
+                            }
+                        })
+                    }
+                })
+                
+            }
+        }
+    }
+    
+}
+
+extension ResultByGroupViewController: UITableViewDelegate {
+}
+
+extension ResultByGroupViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return tests != nil ? tests!.count : 0
```

Comment:
Please do not use force unwrapping.

Reply:
So should I rewrite it with "if let" statement?